### PR TITLE
Update packages

### LINKED ISSUE #110: Update package dependencies to recent versions
A lot of dependencies are outdated, and are not pinned to an exact version.

### PR DESCRIPTION
**Pull request by rmjordas**
**TLDR (Too long; Didn't review)**:

- updated packages
- fix errors caused by package updates

<p>
<details><summary><b>Package updates</b></summary>
<p>

- **`@material-ui/core`: ^1.3.1 -> 3.7.1**
- `@types/jest`: ^23.1.3 -> 23.3.12
- `@types/node`: ^10.5.0 -> 10.12.18
- `@types/react`: ^16.4.4 -> 16.7.18
- `@types/react-dom`: ^16.0.6 -> 16.0.11
- `@types/react-router-dom`: ^4.2.7 -> 4.3.1
- `husky`: ^1.0.0-rc.13 -> 1.3.1
- **`lint-staged`: ^7.2.0 -> 8.1.0**
- `prettier`: 1.14.2 -> 1.15.3
- `react`: ^16.4.1 -> 16.7.0
- `react-dom`: ^16.4.1 -> 16.7.0
- `sendbird`: ^3.0.77 -> 3.0.92
- **`styled-components`: ^3.3.3 -> 4.1.3**
- **`typescript`: ^2.9.2 -> 3.2.2**

</p>
</details>
</p>

<p>
<details><summary><b>New packages</b></summary>
<p>

- `@types/styled-components`: 4.0.3
- `react-scripts`: 2.1.2
- `tslint`: 5.12.0
- `tslint-config-prettier`: 1.17.0
- `tslint-react`: 3.6.0

</p>
</details>
</p>

<p>
<details><summary><b>Removed packages</b></summary>
<p>

- ~~`react-scripts-ts`: 2.16.0~~

</p>
</details>
</p>

<p>
<details><summary><b>Details</b></summary>
<p>

- Migrate from `react-script-ts` to `react-script@2.x`
- Update packages to latest version
- Remove `RouteProps` in component definitions
- Remove unused class property
- Fix type check errors
- Update import paths
- ~~Remove SASS to CSS compilation on CI scripts~~
- ~~Add Sass Lint extension recommendation for VS Code editor~~
- ~~Fix application name in `index.html` and `manifest.json`~~
- ~~Add type safe `styled`  export~~
- ~~Add content for "Environment variables" section in Readme~~
- ~~Change filename to PascalCase if it exports a React component and camelCase for anything else~~
- ~~Remove Redux related code~~
- ~~Fixed `styled` import paths~~

</p>
</details>
</p>

<p>
<details><summary><b>Readings</b></summary>
<p>

- [`create-react-app` docs - Adding TypeScript](https://facebook.github.io/create-react-app/docs/adding-typescript)
- [React blog - Create React App 2.0: Babel 7, Sass, and More ](https://reactjs.org/blog/2018/10/01/create-react-app-v2.html)
- [Migrating from create-react-app-typescript to Create React App](https://vincenttunru.com/migrate-create-react-app-typescript-to-create-react-app/)
- [facebook/create-react-app - TypeScript support using Babel 7](https://github.com/facebook/create-react-app/pull/4837)
- [Styled Components - TypeScript](https://www.styled-components.com/docs/api#typescript)
- [@types/styled-components - 4.1.0 breaks themes](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/30536)
- [Airbnb React/JSX Style Guide](https://github.com/airbnb/javascript/tree/master/react)
- [How to use Sass and CSS Modules with create-react-app](https://blog.bitsrc.io/how-to-use-sass-and-css-modules-with-create-react-app-83fa8b805e5e)

</p>
</details>
</p>

Closes #110

### CODE REVIEW

File: src/components/shared/models/mentorship.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export interface IMentorship {
   createdAt?: string;
   userRatingSubmitted?: boolean;
   mentorRatingSubmitted?: boolean;
+  [key: string]: string | IMentor | IUser | boolean | undefined;
```
Comment (roymoran) [inline src/components/shared/models/mentorship.ts]: What's the value in this line here?

File: src/components/auth/MentorLogin.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { JwtTokenHelper } from '../helpers/jwtTokenHelper';
 import { IJwtAuthToken, ILinkedInAuthCode, IMentor } from '../shared';
 // TODO: Validate state
 
-class MentorLogin extends React.Component<RouteComponentProps<RouteProps>> {
```
Comment (roymoran) [inline src/components/auth/MentorLogin.tsx]: Is this line of code `RouteComponentProps<RouteProps>` === `RouteComponentProps` so including RouteProps is redundant?

File: src/components/chat/Conversation.tsx [hunk 3]
```diff
@@ -96,7 +96,6 @@ export interface IConversationState {
 }
 
 export class Conversation extends React.Component<IConversationProps, IConversationState> {
-  public messageListContainerRef: React.RefObject<HTMLDivElement>;
```
Comment (roymoran) [inline src/components/chat/Conversation.tsx]: I think this line was unused correct?

Comment (rmjordas) [reply src/components/chat/Conversation.tsx]: Yes, This throws a `TypeError`:

```
Property 'messageListContainerRef' has no initializer and is not definitely assigned in the constructor.  TS2564
````

File: src/components/auth/MentorLogin.tsx (hunk 2 above)
Comment (rmjordas) [reply src/components/auth/MentorLogin.tsx]: I think this is due to upgrading `@types/react-router`. This is the error:

```
Type 'RouteProps' does not satisfy the constraint '{ location?: string | undefined; component?: string | undefined; render?: string | undefined; children?: string | undefined; path?: string | undefined; exact?: string | undefined; sensitive?: string | undefined; strict?: string | undefined; }'.
  Types of property 'location' are incompatible.
    Type 'Location<any> | undefined' is not assignable to type 'string | undefined'.
      Type 'Location<any>' is not assignable to type 'string'.  TS2344
```

File: src/components/shared/models/mentorship.ts (hunk 1 above)
Comment (rmjordas) [reply src/components/shared/models/mentorship.ts]: Hmm. I remember having errors with accessing properties dynamically on objects that are typed with `IMentorship`. When I removed this line there weren't any errors anymore. Strange...

I'll remove this change.
